src/Trees/Formula.hs

```diff
@@ -35,3 +37,6 @@ instance Formula (SynTree BinOp Char) where
       applyMaybe _ _ Nothing = Nothing
       applyMaybe f (Just x) (Just y) = Just $ f x y
 
+hasUnusedAtoms :: SynTree BinOp Char -> Bool
+hasUnusedAtoms t = any (\(F.Literal c) -> isSemanticEqual t (replaceSubFormula (Leaf c) (Not (Leaf c)) t)) atoms
+  where atoms = atomics t
```

Comment:
I don't like the fact that `-Wno-incomplete-uni-patterns` is needed above.

As an alternative:
```haskell
hasUnusedAtoms t = any (\c -> isSemanticEqual t (replaceSubFormula (Leaf c) (Not (Leaf c)) t)) atoms
  where atoms = nubOrd (collectLeaves t)
```